core/src/main/java/org/jdbi/v3/core/Jdbi.java

```diff
@@ -332,6 +331,18 @@ public HandleCallbackDecorator getHandleCallbackDecorator() {
         return this.handleCallbackDecorator.get();
     }
 
+    /**
+     * Returns an internal handle supplier holder element used by this Jdbi instance.
+     * <br>
+     * <b>This is an internal method and not part of the public API!</b>
+     * @return A handle supplier holder element
+     * @param <T> type parameter
+     */
+    @Alpha
+    public <T extends HandleSupplierHolder> T getHandleSupplierHolder() {
```

Comment:
Since this is not intended as public API, can we restrict visibility? Also, I think `@Alpha` does not need to be applied to non-public api, since it is always subject to change at any time.

Reply:
Unfortunately not. This is specifically intended to be used by the Kotlin code to allow doing this:

```kotlin
fun Jdbi.attachToContext(): CoroutineContext.Element = this.getHandleSupplierHolder<CoroutineAwareHandleSupplierHolder>().contextElement()
```

This allows attaching the Jdbi object to a coroutine context so that the handles in the internal threadlocal are moved around correctly. 

The extension function above is (in terms of java code) a hidden static function (so basically it is 

```java
public static CoroutineContext.Element attachToContext() {
    return Jdbi<CoroutineAwareHandleSupplierHolder>getHandleSupplierHolder().contextElement();
}
```

 "patched" into the Jdbi class. And that is only possible with public members (because the extension method lives in a different package (org.jdbi.v3.core.kotlin) and is not an actual member of the hierarchy, so neither protected nor package scope would work).